File: 2323-MinimumBitFlipsToConvertNumber/2323-MinimumBitFlipsToConvertNumber.cpp
// Last updated: 04/04/2026, 23:30:19
class Solution {
public:
    int minBitFlips(int start, int goal) {
        int ans = 0;
        int XOR = start^goal;
        while(XOR>0) {
            ans+=(XOR&1);
            XOR>>=1;
        }
        return ans;
    }
};

File: 2323-MinimumBitFlipsToConvertNumber/2323-MinimumBitFlipsToConvertNumber.java
// Last updated: 04/04/2026, 23:30:22
class Solution {
    public int minBitFlips(int start, int goal) {
        int xor=(start ^ goal);
        int count=0;
        while(xor != 0) {
            count = count + (xor & 1);
            xor = xor>>1;
        }
        return count;
    }
}